lms/static/js/learner_dashboard/models/course_card_model.js

```diff
@@ -15,19 +15,46 @@
                 }
             },
 
+            getUnselectedRunMode: function(runModes) {      
+                if(runModes && runModes.length > 0){  
+                    return {
+                        course_image_url: runModes[0].course_image_url,
+                        marketing_url: runModes[0].marketing_url,
+                        is_enrollment_open: runModes[0].is_enrollment_open,
+                        enrollment_open_date: runModes[0].enrollment_open_date
+                    };
+                }
+                return {};
+            },
+
             getRunMode: function(runModes){
-                //we should populate our model by looking at the run_modes
-                if (runModes.length > 0){
-                    if(runModes.length === 1){
-                        return runModes[0];
+                var enrolled_mode = _.findWhere(runModes, {is_enrolled: true}),
+                    openEnrollmentRunModes = this.getEnrollableRunModes();
+                //we populate our model by looking at the run_modes
+                if (enrolled_mode){
+                    // If we have a run_mode we are already enrolled in,
+                    // return that one always
+                    return enrolled_mode;
+                }
+
+                if(openEnrollmentRunModes.length > 0){
```

Comment:
For clarity it would make sense to have this be an ```else if``` from the previous if statement.